install.sh

```diff
@@ -71,22 +115,64 @@ main () {
         exit 1
     fi
 
-    URL="https://github.com/opengrep/opengrep/releases/download/${VERSION}/${DIST}"
-    echo
-    echo "*** Installing Opengrep ${VERSION} for ${OS} (${ARCH}) ***"
-    echo
+    if [ "$REQUESTED_VERSION" = "latest" ]; then
+        # Use GitHub's latest release redirect
+        URL="https://github.com/opengrep/opengrep/releases/latest/download/${DIST}"
+
+        # Get the final version from the redirect
+        REDIRECT_URL=$(curl -sI "$URL" 2>/dev/null | grep -i '^location:' | sed 's/location: //i' | tr -d '\r\n' || true)
+        if [ -n "$REDIRECT_URL" ]; then
+            # Extract version from the redirect URL
+            FINAL_VERSION=$(echo "$REDIRECT_URL" | sed -E 's|.*/download/([^/]+)/.*|\1|')
+        else
+            echo "Failed to resolve latest version. The redirect from $URL did not work." 1>&2
+            echo "Please check your internet connection or specify a version with -v" 1>&2
+            exit 1
+        fi
+
+        echo
+        echo "*** Installing Opengrep ${FINAL_VERSION} for ${OS} (${ARCH}) ***"
+        echo
+    else
+        URL="https://github.com/opengrep/opengrep/releases/download/${REQUESTED_VERSION}/${DIST}"
+        echo
+        echo "*** Installing Opengrep ${REQUESTED_VERSION} for ${OS} (${ARCH}) ***"
+        echo
+    fi
+
+    # Now set INST with the final version
+    INST="${PREFIX}/${FINAL_VERSION}"
 
     # check if binary already exists
     if [ -f "${INST}/opengrep" ]; then
         echo "Destination binary ${INST}/opengrep already exists."
     else
+        # Download to temp file first
+        TEMP_FILE=$(mktemp)
+        trap 'rm -f $TEMP_FILE' EXIT
```

Comment:
It's not very likely that a download will be interrupted in the middle, especially in a cloud environment.

What motivated you to add this?

Reply:
The updated script cleans up any files on error, so this is not needed.